controllers/istiorevisiontag/istiorevisiontag_controller.go

```diff
@@ -0,0 +1,571 @@
+	revision := getReferencedRevisionTagFromPod(pod.GetLabels())
+	if revision != "" {
+		return []reconcile.Request{{NamespacedName: types.NamespacedName{Name: revision}}}
+	}
+	return nil
+}
+
+func (r *Reconciler) mapOperatorResourceToReconcileRequest(ctx context.Context, obj client.Object) []reconcile.Request {
+	var revisionName string
+	if i, ok := obj.(*v1alpha1.Istio); ok && i.Status.ActiveRevisionName != "" {
+		revisionName = i.Status.ActiveRevisionName
+	} else if rev, ok := obj.(*v1alpha1.IstioRevision); ok {
+		revisionName = rev.Name
+	} else {
+		return nil
+	}
+	tags := v1alpha1.IstioRevisionTagList{}
+	labelSelector := map[string]string{
+		sailOperatorReferencedRevisionLabel: revisionName,
+	}
+	err := r.Client.List(ctx, &tags, &client.ListOptions{LabelSelector: labels.SelectorFromSet(labelSelector)})
+	if err != nil {
+		return nil
+	}
+	requests := []reconcile.Request{}
+	for _, revision := range tags.Items {
+		requests = append(requests, reconcile.Request{NamespacedName: types.NamespacedName{Name: revision.Name}})
+	}
+	return requests
+}
+
+// ignoreStatusChange returns a predicate that ignores watch events where only the resource status changes; if
+// there are any other changes to the resource, the event is not ignored.
+// This ensures that the controller doesn't reconcile the entire IstioRevisionTag every time the status of an owned
+// resource is updated. Without this predicate, the controller would continuously reconcile the IstioRevisionTag
+// because the status.currentMetrics of the HorizontalPodAutoscaler object was updated.
+func ignoreStatusChange() predicate.Funcs {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return specWasUpdated(e.ObjectOld, e.ObjectNew) ||
+				!reflect.DeepEqual(e.ObjectNew.GetLabels(), e.ObjectOld.GetLabels()) ||
+				!reflect.DeepEqual(e.ObjectNew.GetAnnotations(), e.ObjectOld.GetAnnotations()) ||
+				!reflect.DeepEqual(e.ObjectNew.GetOwnerReferences(), e.ObjectOld.GetOwnerReferences()) ||
+				!reflect.DeepEqual(e.ObjectNew.GetFinalizers(), e.ObjectOld.GetFinalizers())
+		},
+	}
+}
+
+func specWasUpdated(oldObject client.Object, newObject client.Object) bool {
+	// for HPAs, k8s doesn't set metadata.generation, so we actually have to check whether the spec was updated
+	if oldHpa, ok := oldObject.(*autoscalingv2.HorizontalPodAutoscaler); ok {
+		if newHpa, ok := newObject.(*autoscalingv2.HorizontalPodAutoscaler); ok {
+			return !reflect.DeepEqual(oldHpa.Spec, newHpa.Spec)
+		}
+	}
+
+	// for other resources, comparing the metadata.generation suffices
+	return oldObject.GetGeneration() != newObject.GetGeneration()
+}
+
+func wrapEventHandler(logger logr.Logger, handler handler.EventHandler) handler.EventHandler {
+	return enqueuelogger.WrapIfNecessary(v1alpha1.IstioRevisionTagKind, logger, handler)
+}
+
+func UnpackReconcilerError(err error) error {
```

Comment:
This is basically what [Unwrap](https://pkg.go.dev/errors#Unwrap) does?

Reply:
Good point. Using `Unwrap()` now